subprojects/gradle/build-verdict/src/gradleTest/kotlin/com/avito/android/build_verdict/BuildVerdictPluginConfigurationPhaseTest.kt

```diff
@@ -51,10 +51,20 @@ class BuildVerdictPluginConfigurationPhaseTest : BaseBuildVerdictTest() {
             expectFailure = true
         )
         result.assertThat().buildFailed()
-        assertBuildVerdictFiles(
-            expectedPlainTextVerdict = plainTextVerdicts.illegalMethodFails(),
-            expectedHtmlVerdict = htmlVerdicts.illegalMethodFails()
-        )
+
+        val agpVersion = System.getProperty("androidGradlePluginVersion")
```

Comment:
Is it possible to detect it from AGP itself?
I wonder how other tools do it.

Reply:
I don't know, but here, on test code level, we only have a classpath to analyze and already defined system property
i prefered second one

Reply:
I see this as a common issue to many plugins based on AGP.
It seems like a feature request or at least a question to Google.

Reply:
yes, the main problem is i can test only with AGP version provided from plugin classpath;
changing agp version in TestProjectGenerator does nothing

I see it could be handled with multiple gradle test configurations for plugin

Reply:
Take a look at `com.android.Version.ANDROID_GRADLE_PLUGIN_VERSION`. Is it what we need?

Reply:
Thanks, looks like it

![image](https://user-images.githubusercontent.com/1105133/123456526-9cd27080-d5eb-11eb-96d4-51c62315dcf7.png)

Reply:
https://github.com/avito-tech/avito-android/pull/1104